### ISSUE TITLE
Support sdk 32

### ISSUE THREAD

**Issue by anthony-fresneau-kiplin** [robolectric/robolectric#6920]
### Description
An error occurred when targetSdkVersion is 32 : 
Package targetSdkVersion=32 > maxSdkVersion=31
### Steps to Reproduce
define         targetSdkVersion 32 in build.gradle
### Robolectric & Android Version
roboelectric 4.7.3
targetSdkVersion 32
compileSdkVersion 32

**Comment by utzcoz** [comment]
@anthony-fresneau-kiplin thanks for reporting this problem so early after Android 12L beta1 released. Robolectric will support it after Android 12L source code released with stable SDK 32.

**Comment by utzcoz** [comment]
> @anthony-fresneau-kiplin thanks for reporting this problem so early after Android 12L beta1 released. Robolectric will support it after Android 12L source code released with stable SDK 32.

Android 12L source code released finally. Hi @hoisie , is there any plan to land supporting for SDK 32?

**Comment by hoisie** [comment]
Yeah it will happen very soon, I'll start a code sync today.

**Comment by utzcoz** [comment]
@hoisie What will Robolectric do after Android 12 QPR3 released(https://arstechnica.com/gadgets/2022/03/android-now-has-stable-beta-and-dev-channels-just-like-chrome/)? Looks like Android 12L is QPR2, and now there is a QPR3 before Android 13 released. I think Android 13 will take SDK 33, and Android 12 QPR3 will keep SD 32, is that right? If that is correct, will Robolectric update android-all and instrumented android-all after QPR3 released? Or Robolectric will keep using the same android-all and instrumented android-all as Android 12L? If I misunderstood something, please let me know. The new release cycle is new to me, and I'm learning what Android team want to do for this release cycle from online materials.

**Comment by hoisie** [comment]
@utzcoz to be honest this is very confusing for me too. For the purposes of Robolectric I hope it doesn't matter. I am planning to pull a recent build from the sc-v2-release. At this point in the cycle, there are a relatively small number of changes going into sc-v2-release, only to fix very high priority bugs. Also many of these changes don't touch `frameworks/base`, which is the only part of the platform that is relevant to Robolectric.

**Comment by utzcoz** [comment]
@hoisie Hope so. And could you document how to build android-all and instrumented android-all jars from AOSP source code on Wiki? I also have interested in process to build those things.

**Comment by utzcoz** [comment]
The PR to support SDK 32 has been merged, @hoisie will we close it after release? Or will we close it immediately? I remember you prefer to the first one.

**Comment by hoisie** [comment]
@utzcoz just put up https://github.com/robolectric/robolectric/wiki/Building-framework-JARs-from-platform-sources

TLDR is `m -j robolectric-host-android_all`

https://cs.android.com/android/platform/superproject/+/master:external/robolectric-shadows/Android.bp;l=112

Also https://github.com/robolectric/robolectric/wiki/Update-Preinstrumented-Jars-to-a-new-version

Let me know if you'd like any more info about it :)

I will try to release 4.8-alpha-1 tomorrow

**Comment by utzcoz** [comment]
@hoisie Thanks. That's enough for me.  :)

**Comment by utzcoz** [comment]
Hi @anthony-fresneau-kiplin, Robolectric 4.8-alpha1 has been released, and it brings supporting for SDK 32, what about trying it?

**Comment by anthony-fresneau-kiplin** [comment]
Hi @utzcoz  , i have just tried the 4.8-alpha1 and the support of SDK32 is OK. Thanks for the support. Will it be release soon?

**Comment by utzcoz** [comment]
> Hi @utzcoz , i have just tried the 4.8-alpha1 and the support of SDK32 is OK. Thanks for the support. Will it be release soon?

@anthony-fresneau-kiplin Thanks for testing it. I think @hoisie can help to provide a detailed release timeline for 4.8.

**Comment by jaredsburrows** [comment]
@hoisie Any update on this?

**Comment by utzcoz** [comment]
@jaredsburrows @anthony-fresneau-kiplin The 4.8 released with SDK 32 supporting. Please check this new stable version.

**Comment by anthony-fresneau-kiplin** [comment]
Yeah, it's working. Thanks

**Comment by shamy91** [comment]
Roboelectric needs to be updated to support sdk 33

**Comment by utzcoz** [comment]
> Roboelectric needs to be updated to support sdk 33

Robolectric 4.9 has supported SDK 33.
